src/lib/LED/LED.h

```diff
@@ -23,104 +23,143 @@ void WS281BsetLED(uint32_t color)
+    {
+        return lightness << 16 | lightness << 8 | lightness;
+    }
+
+    region = hue / 43;
+    remainder = (hue - (region * 43)) * 6; 
+
+    p = (lightness * (255 - saturation)) >> 8;
+    q = (lightness * (255 - ((saturation * remainder) >> 8))) >> 8;
+    t = (lightness * (255 - ((saturation * (255 - remainder)) >> 8))) >> 8;
+
+    switch (region)
+    {
+        case 0:
+            return lightness << 16 | t << 8 | p;
+        case 1:
+            return q << 16 | lightness << 8 | p;
+        case 2:
+            return p << 16 | lightness << 8 | t;
+        case 3:
+            return p << 16 | q << 8 | lightness;
+        case 4:
+            return t << 16 | p << 8 | lightness;
+        default:
+            return lightness << 16 | p << 8 | q;
+    }
+}
+
+static void blinkyUpdate(unsigned long now) {
+    if (blinkyUpdateTime > now)
+    {
+        return;
+    }
+
+    uint32_t rgb = HsvToRgb(hue, saturation, lightness);
+    WS281BsetLED(rgb);
+    DBGVLN("LED hue %u", hue);
+    if ((int)hue + hueStepValue > 255) {
+        if ((int)lightness - lightnessStep < 0) {
+            state = NORMAL;
+            return;
+        }
+        lightness -= lightnessStep;
+    } else {
+        hue += hueStepValue;
+    }
+    blinkyUpdateTime = now + 3000/(256/hueStepValue);
+}
+
+void startupLEDs()
+{
+    WS281Binit();
+    state = STARTUP;
+    #ifdef PLATFORM_ESP32
+    // Only do the blinkies if it was NOT a software reboot
+    if (esp_reset_reason() == ESP_RST_SW) {
+        state = NORMAL;
+    }
+    #endif
+    blinkyUpdateTime = 0;
+    hue = 0;
+    lightness = 128;
 }
-#endif
 
 void updateLEDs(uint32_t now, connectionState_e connectionState, uint8_t rate, uint32_t power)
 {
-#if (defined(PLATFORM_ESP32) && defined(GPIO_PIN_LED)) || defined(WS2812_LED_IS_USED)
-    constexpr uint32_t rate_colors[RATE_MAX] =
+    if (state == STARTUP)
     {
-        0x0000FF,     // 500/200 hz  blue
-        0x00FF00,     // 250/100 hz  green
-        0xFF8000,     // 150/50 hz   orange
-        0xFF0000      // 50/25 hz    red
+        blinkyUpdate(now);
+        return;
+    }
+
+    constexpr uint8_t rate_hue[RATE_MAX] =
+    {
+        170,     // 500/200 hz  blue
+        85,     // 250/100 hz  green
+        21,     // 150/50 hz   orange
+        0      // 50/25 hz    red
     };
-    constexpr uint32_t LEDupdateInterval = 100;
-    static uint8_t LEDfade = 0;
-    static int8_t LEDfadeAmount = 2;
-    static uint32_t LEDupdateCounterMillis;
+    constexpr uint32_t LEDupdateInterval = 50;
+    static uint32_t nextUpdateTime = 0;
 
     static connectionState_e lastState = disconnected;
-    static uint32_t lastColor = 0xFFFFFFFF;
+    static uint8_t lastRate = 0xFF;
     static uint32_t lastPower = 0xFFFFFFFF;
 
-    uint32_t color = rate_colors[rate];
-    if ((connectionState == disconnected) && (now - LEDupdateCounterMillis > LEDupdateInterval))
+    if ((connectionState == disconnected) && (now > nextUpdateTime))
```

Comment:
`now > nextUpdateTime` fails on overflow. Comparison should **always** be `now - last [comparison] interval` to avoid overflow errors.